### DISCUSSION TITLE
`str(JsProxy)` raising `TypeError`

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [pyodide/pyodide#4569] [Q&A] [answered]
When I try to `str()` a `JsProxy` in python scope, a `TypeError` occurs that:

```py
TypeError: Pyodide cannot generate a repr for this Javascript object because it has no 'toString' method
```

Note that the `JsProxy` is a `Module` object.

### To Reproduce

```py
>>> from pyodide.code import run_js
>>> a_simple_module = await run_js("import('https://esm.run/partial-json')")
>>> str(a_simple_module)
```

### Expected behavior

```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: Pyodide cannot generate a repr for this Javascript object because it has no 'toString' method
```

### Environment

the `console.html` from the [pyodide docs](https://pyodide.org/en/stable/console.html).

**Comment by CNSeniorious000** [comment]
I know a `Module` object can't be `.toString()` or `String()` in JavaScript, but `console.log` always works. I wonder how to generate a string representation for a JavaScript object promisingly.

I think, in python, `str()` shouldn't raise an error, usually.

**Comment by hoodmane** [comment, marked as answer]
Thanks for the report! We probably should fall back to `Object.prototype.toString.call` in this case. That would render it as `'[object Module]'` which isn't useful persay but at least doesn't throw. @ryanking13 wdyt?

To avoid the request, a better repro is:
```py
from pyodide.code import run_js

mod = await run_js("import('data:text/javascript,')")
print(mod)
```

**Reply by CNSeniorious000** [reply]
Apologies for my lack of familiarity with the import syntax. I support this approach too.

**Reply by ryanking13** [reply]
Yes, I agree that `str` and `repr` should success always even if the result is not very helpful. Using `Object.prototype.toString.call` seems reasonable to me.
